rslib/src/import_export/mod.rs

```diff
@@ -81,3 +85,14 @@ impl ImportError {
         .into()
     }
 }
+
+static LAST_IMPORT_RESPONSE: Lazy<Mutex<ImportResponse>> =
+    Lazy::new(|| Mutex::new(ImportResponse::default()));
+
+pub fn set_last_import_response(import_response: &ImportResponse) {
+    *LAST_IMPORT_RESPONSE.lock().unwrap() = import_response.clone();
+}
+
+pub fn get_last_import_response() -> ImportResponse {
+    LAST_IMPORT_RESPONSE.lock().unwrap().clone()
+}
```

Comment:
It's better to avoid keeping state when possible - otherwise this sits around consuming memory even after the importing dialog has closed, and makes it harder to reason about the code.

An alternative you could use is to shift the importAnkiPackage() call into the webpage, like how the CSV importer invokes importCsv(). That way the import log is returned to the webpage, and can be displayed without it ever having to be saved somewhere else.

It might be nice if in the future we had a similar page shown when importing a CSV - perhaps the current log could be adapted in the future to handle either csv or apkg?

Reply:
Ah, I was sure there is a cleaner solution that eluded me.

The page is already shown for CSV imports.

Reply:
Nice!

Currently this seems to involve closing the csv import window and opening a new window to show the log. Since both apkg and csv imports will be invoked via a webpage now, it would be nice to update the existing webview instead of creating a new one. I imagine it could be done with something like:

```
{{#if importResponse}} <ImportLog {importResponse} /> {:else} <ImportCsvPage ...> {/if}
```

If moving the data fetching from import-csv/index.ts into ImportCsvPage makes things easier, no objections there.

In the apkg case, since we don't have anything to present to the user, we could just show a spinner perhaps?

Reply:
I think it'd be bad to lose informative progress reporting if we only show a simple spinner. Maybe backend progress handling in progress.py can be re-implemented on the web side? I'll look into that.

Reply:
You're right, I'd overlooked that. But it should be possible to show the progress in TypeScript; that's one of the reasons why I made it work via poll instead of a callback.